src/openloco/ui/dropdown.cpp

```diff
@@ -263,43 +725,162 @@ namespace openloco::ui::dropdown
      * count @<bl>
      * flags @<bh>
      */
-    void show_text_2(int16_t x, int16_t y, int16_t width, int16_t height, colour_t colour, size_t count, uint8_t flags)
+    void showText2(int16_t x, int16_t y, int16_t width, int16_t height, uint8_t itemHeight, colour_t colour, size_t count, uint8_t flags)
     {
         assert(count < std::numeric_limits<uint8_t>::max());
 
-        registers regs;
-        regs.cx = x;
-        regs.dx = y;
-        regs.al = colour;
-        regs.bl = static_cast<uint8_t>(count);
-        regs.bh = flags;
-        regs.bp = width;
-        regs.di = height;
+        common::setColourAndInputFlags(&colour, &flags);
+
+        _dropdownColumnCount = 1;
+        _dropdownItemWidth = width;
+        _dropdownItemHeight = 10;
+
+        if (flags & (1 << 6))
+        {
+            _dropdownItemHeight = itemHeight;
+        }
+
+        flags &= ~(1 << 6);
+
+        _dropdownItemCount = (uint16_t)count;
+        _dropdownRowCount = (uint32_t)count;
+        common::widgets[0].colour = colour;
+        uint16_t dropdownHeight = (uint16_t)count * _dropdownItemHeight + 3;
+        common::widgets[0].bottom = dropdownHeight;
+        dropdownHeight++;
+
+        gfx::ui_size_t size = { (uint16_t)width, (uint16_t)height };
+        gfx::point_t origin = { x, y };
+        origin.y += height;
+
+        size.height = dropdownHeight;
+        if ((size.height + origin.y) > ui::height() || origin.y < 0)
+        {
+            origin.y -= (height + dropdownHeight);
+            auto dropdownBottom = origin.y;
+
+            if (origin.y >= 0)
+            {
+                dropdownBottom = origin.y + dropdownHeight;
+            }
+
+            if (origin.y < 0 || dropdownBottom > ui::height())
+            {
+                origin.x += width + 3;
+                origin.y = 0;
+            }
+        }
+
+        size.width = width + 3;
+        common::widgets[0].right = size.width;
+        size.width++;
+
+        if (origin.x < 0)
+        {
+            origin.x = 0;
+        }
+
+        origin.x += size.width;
+
+        if (origin.x > ui::width())
+        {
+            origin.x = ui::width();
+        }
+
+        origin.x -= size.width;
 
-        call(0x4CCC7C, regs);
+        common::open(origin, size, colour);
+    }
+
+    void showText2(int16_t x, int16_t y, int16_t width, int16_t height, colour_t colour, size_t count, uint8_t flags)
+    {
+        showText2(x, y, width, height, 0, colour, count, flags & ~(1 << 6));
     }
 
     // 0x004CF2B3
     void populateCompanySelect(window* window, widget_t* widget)
     {
-        registers regs;
-        // regs.edx = widgetIndex;
-        regs.edi = (int32_t)widget;
-        regs.esi = (int32_t)window;
+        char* buffer = _byte_112CC04;
```

Comment:
in this function 112CC04 is just being used as an array of bools. Just make a new array of bools.
`std::array<bool, 16> companyOrdered = {};`
That way you don't need to clear the array first.